src/couch_replicator.erl

```diff
@@ -332,6 +334,21 @@ do_init(#rep{options = Options, id = {BaseId, Ext}, user_ctx=UserCtx} = Rep) ->
         }
     }.
 
+% If max http connection is 1 and an http source is used,
+% a deadlock could occur. Changes feed could consume a worker
+% from the pool, then makes a gen_server call back to
+% replication process. In the meantime the main replication process could
+% be calling get_pending_changes and in there wait for an available http
+% connection.
+%
+% see FB-61178
+adjust_maxconn(1, #httpdb{}, RepId) ->
+    Msg = "Adjusting max http connections for replication `~p` to 2",
```

Comment:
Might make this more specific: "Minimum number of HTTP source connections is two, adjusting automatically." Or something to that effect.